### PR TITLE
[ContactStructuralMechanicsApplication] Transition PR improving frictional contact

### PR DESCRIPTION
**Pull request by loumalouomega**
This fixes many things, still not as robust as I would wish. Working on that (basically is a problem of the tangent component, which flips if the simulation is done faster than a certain threshold, this is not robust :P). this PR fix several things, refactors others and add several tests. 

Additionally reduces the size of the condition, but probably this will be reverted in the final version, as this reduction is due to a temporal simplification in the tangent computation (the idea should be to the tangent stress to be consistent with the slip, thing that doesn¡t happen in more complex problems)

### CODE REVIEW

**Comment by loumalouomega** [comment]
BTW, my app reached 99 python tests :D, one more and I will break the barrier of the 100

File: applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_weighted_gap.cpp [hunk 1]
```diff
@@ -922,7 +922,7 @@ namespace Kratos
 
             const double tolerance = 1.0e-4;
             array_1d<double, 3> slip = ZeroVector(3);
-            slip[0] = delta_x;
+            slip[0] = - delta_x;
```
Comment (AlejandroCornejo) [inline applications/ContactStructuralMechanicsApplication/tests/cpp_tests/test_weighted_gap.cpp:925]: uhmm jeje

**Review by AlejandroCornejo — APPROVED** [review]
nothing weird in my opinion

**Comment by loumalouomega** [comment]
Yep :P

**Pull request merged**